src/renderer/src/components/notice/Notice.tsx

```diff
@@ -1,108 +1,155 @@
+    pausedTime = Date.now() - startTime;
+  };
 
-  const onRef = (notice: HTMLElement) => {
-    props.updateGradient.pulse();
-    props.onMount(notice);
+  const resumeRemoveTimeout = () => {
+    startTime = Date.now() - pausedTime;
+    timeout = setTimeout(removeNotice, NOTICE_DURATION - pausedTime);
   };
 
+  onMount(() => {
+    setTimeout(() => setIsVisible(true), 50); // Delay to trigger enter animation
+    startRemoveTimeout();
+  });
+
+  createEffect(() => {
+    clearTimeout(timeout);
+    if (props.isPaused) {
+      pauseRemoveTimeout();
+    } else {
+      resumeRemoveTimeout();
+    }
+  });
+
   return (
     <div
-      class={"notice-wrapper"}
-      onPointerEnter={pauseDrain}
-      onPointerLeave={() => setDrainTime(props.notice.timeoutMS ?? 10_000)}
+      ref={noticeRef}
+      class={twMerge(
+        bgStyle({ variant: props.notice.variant }),
+        `after:absolute after:inset-0 after:-z-20 after:size-20 after:rounded-full after:blur-2xl after:content-['']`,
+        `before:absolute before:inset-0.5 before:-z-10 before:rounded-[10px] before:bg-thick-material before:content-['']`,
+        "group relative w-96 transform overflow-hidden rounded-xl bg-thick-material p-4 shadow-2xl ring-2 ring-inset ring-stroke backdrop-blur-md duration-300 ease-in-out",
+        isVisible() ? "translate-y-0 opacity-100" : "-translate-y-full opacity-0 blur-sm",
+        isRemoving()
+          ? "my-0 h-0 max-h-0 min-h-0 -rotate-12 scale-75 py-0 opacity-0 blur-sm"
+          : "my-2 min-h-20",
+      )}
       data-id={props.notice.id}
-      ref={onRef}
     >
-      <div class={"notice " + (props.notice.class !== "notice" ? props.notice.class : "")}>
-        <div class="content">
-          <div class="head">
-            <h3>{props.notice.title}</h3>
-            <button onClick={removeNotice}>
-              <XIcon size={20} />
-            </button>
-          </div>
+      <Button
+        variant="outlined"
+        size="icon"
+        onClick={removeNotice}
+        class="absolute right-3 top-3 size-5 p-1 text-subtext opacity-0 transition-opacity duration-150 group-hover:opacity-100"
+      >
+        <XIcon size={16} />
+      </Button>
 
-          <p>{props.notice.content}</p>
+      <div class="mr-6 flex items-start">
+        <Show when={props.notice.icon}>
+          <div
+            class={twMerge(
+              "mr-3 mt-0.5 flex-shrink-0",
+              textStyle({ variant: props.notice.variant }),
+            )}
+          >
+            {props.notice.icon}
+          </div>
+        </Show>
+        <div class="overflow-hidden">
+          <Show when={props.notice.title}>
+            <h3 class="mb-1 overflow-hidden text-ellipsis whitespace-nowrap text-base font-semibold">
+              {props.notice.title}
+            </h3>
+          </Show>
+          <Show when={props.notice.description}>
+            <p class="overflow-hidden text-ellipsis whitespace-nowrap text-sm text-subtext">
```

Comment:
I don't think we should limit the number of lines on the notice description to only one line, 2 lines should be good. (`line-clamp-2`)